services/garage-backend/src/services/thingService.ts

```diff
@@ -17,7 +15,39 @@ function findById(thingId: string): Promise<ThingDocument> {
     });
 }
 
-function findAll(): Promise<ThingDocument[]> {
+function findAllThings(): Promise<ThingDocument[]> {
   return Thing.find().sort({ name: 1, descripton: -1 }).exec();
 }
-*/
+
+function updateThing(
+  thingId: string,
+  update: Partial<ThingDocument>,
+): Promise<ThingDocument | null> {
+  return Thing.findByIdAndUpdate(thingId, update, { new: true })
+    .exec()
+    .then((thing) => {
+      if (!thing) {
+        throw new Error(`Movie ${thingId} not found`);
```

Comment:
```suggestion
        throw new Error(`Thing ${thingId} not found`);
```
probably not a movie 😄